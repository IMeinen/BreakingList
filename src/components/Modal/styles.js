import styled from 'styled-components';
import { MdClose } from 'react-icons/md';

export const Container = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;
  background: rgb(17, 17, 17);
  background: linear-gradient(
    0deg,
    rgba(17, 17, 17, 1) 0%,
    rgba(0, 0, 0, 1) 100%
  );

  position: absolute;
  width: 900px;
  height: 700px;
  padding: 40px;

  @media only screen and (max-width: 1000px) {
    width: 90%;
    height: 90vh;
    align-items: center;
  }
  @media only screen and (max-width: 468px) {
    height: 95vh;
  }
  @media only screen and (max-width: 330px) {
    padding: 20px;
        }
  > .core-informations {
    display: flex;
    align-items: flex-start;
    justify-content: flex-start;
    @media only screen and (max-width: 468px) {
      flex-direction: column;
      align-items: center;
    }

    > .image-container {
      width: 360px;
      height: 460px;
      background: #fff;
      @media only screen and (max-width: 468px) {
        width: 230px;
        height: 293px;
      }
      @media only screen and (max-width: 330px) {
          width: 200px;
          height: 254px;
        }
      > img {
        width: 360px;
        height: 460px;
        @media only screen and (max-width: 468px) {
          width: 230px;
          height: 293px;
        }
        @media only screen and (max-width: 330px) {
          width: 200px;
          height: 254px;
        }
      }
    }

    > .description-container {
      width: 90%;
      height: 460px;
      display: flex;
      margin-left: 30px;
      flex-direction: column;
      align-items: flex-start;
      justify-content: space-between;
      color: #fff;

      @media only screen and (max-width: 468px) {
        height: 90%;
        margin: 15px 0px 5px 0px;
      }

      > .description-line {
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        justify-content: flex-start;
        @media only screen and (max-width: 468px) {
          h1 {
            font-size: 16px;
          }
          p {
            font-size: 10px;
          }
        }
      }
    }
  }

  > .footer-informations {
    width: 100%;
    margin-top: 40px;

    color: #fff;
    @media only screen and (max-width: 1000px) {
      margin-top: auto;
    }
    @media only screen and (max-width: 468px) {
      h2 {
        font-size: 12px;
      }
    }
    @media only screen and (max-width: 330px) {
      h2 {
        font-size:8px;
      }
    }

  }
`;

export const StyledClose = styled(MdClose)`
  position: absolute;
  top: 15px;
  right: 15px;
  transition: all 0.3s;
  color: #fff;
  font-size: 56px;

  @media only screen and (max-width: 468px) {
    top: 10px;
    right: 10px;
    font-size: 28px;
  }
  &:hover {
    transform: scale(1.2);
    color: #aaa;
  }
`;
