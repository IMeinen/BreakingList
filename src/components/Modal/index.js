import React from 'react';

import PropTypes from 'prop-types';
import { Container,StyledClose } from './styles';

export default function Modal({ charInfo,modalOpened }) {

  return (
    <Container>
      <StyledClose  onClick={() => modalOpened(false)  }/>
      <div className="core-informations">
        <div className="image-container">
          <img src={charInfo.img} alt={`bg of ${charInfo.char_id}`} />
        </div>
        <div className="description-container">
          <div className="description-line">
            <h1>Nome:</h1>
            <p>{charInfo.name}</p>
          </div>
          <div className="description-line">
            <h1>Aniversário:</h1>
            <p>{charInfo.birthday.split('-').join('/')}</p>
          </div>
          <div className="description-line">
            <h1>Ocupação:</h1>
            <p>{charInfo.occupation.join(', ')}</p>
          </div>
          <div className="description-line">
            <h1>Situação:</h1>
            <p>{charInfo.status}</p>
          </div>
          <div className="description-line">
            <h1>Apelido:</h1>
            <p>{charInfo.nickname}</p>
          </div>
          <div className="description-line">
            <h1>Aparece nas temporadas:</h1>
            <p>{charInfo.appearance.join(', ')}</p>
          </div>
        </div>
      </div>
      <div className="footer-informations">
        <h2>{`O personagem ${charInfo.name} é intrepertado por ${charInfo.portrayed} que atuou nas seguintes séries: ${charInfo.category}.`}</h2>
      </div>
    </Container>
  );
}

Modal.propTypes = {
  charInfo:
    PropTypes.shape({
      appearance: PropTypes.arrayOf(PropTypes.number),
      better_call_saul_appearance: PropTypes.arrayOf(PropTypes.number),
      birthday: PropTypes.string,
      category: PropTypes.string,
      char_id: PropTypes.number,
      img: PropTypes.string,
      name: PropTypes.string,
      nickname: PropTypes.string,
      occupation: PropTypes.arrayOf(PropTypes.string),
      portrayed: PropTypes.string,
      status: PropTypes.string,
    }).isRequired,
  modalOpened: PropTypes.func.isRequired,
};
