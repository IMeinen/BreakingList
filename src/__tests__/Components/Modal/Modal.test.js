import React from 'react';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import fetchData from '../../../__mocks__/Api';
import Modal from '../../../components/Modal';

Enzyme.configure({ adapter: new Adapter() });
describe('Modal component tests', () => {
  it('renders correctly', async () => {
    const mocked = await fetchData();

    if (mocked !== undefined) {
      const ComponentProps =  {
        charInfo: mocked[0],
        modalOpened: console.log,
      };

      const wrapper = shallow(<Modal {...ComponentProps} />);

      expect(wrapper).toMatchSnapshot();
    }
  });
});
