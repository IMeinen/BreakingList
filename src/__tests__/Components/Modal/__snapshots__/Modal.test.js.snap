// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component tests renders correctly 1`] = `
<styled.div>
  <Styled(MdClose)
    onClick={[Function]}
  />
  <div
    className="core-informations"
  >
    <div
      className="image-container"
    >
      <img
        alt="bg of 1"
        src="https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg"
      />
    </div>
    <div
      className="description-container"
    >
      <div
        className="description-line"
      >
        <h1>
          Nome:
        </h1>
        <p>
          Walter White
        </p>
      </div>
      <div
        className="description-line"
      >
        <h1>
          Aniversário:
        </h1>
        <p>
          09/07/1958
        </p>
      </div>
      <div
        className="description-line"
      >
        <h1>
          Ocupação:
        </h1>
        <p>
          High School Chemistry Teacher, Meth King Pin
        </p>
      </div>
      <div
        className="description-line"
      >
        <h1>
          Situação:
        </h1>
        <p>
          Presumed dead
        </p>
      </div>
      <div
        className="description-line"
      >
        <h1>
          Apelido:
        </h1>
        <p>
          Heisenberg
        </p>
      </div>
      <div
        className="description-line"
      >
        <h1>
          Aparece nas temporadas:
        </h1>
        <p>
          1, 2, 3, 4, 5
        </p>
      </div>
    </div>
  </div>
  <div
    className="footer-informations"
  >
    <h2>
      O personagem Walter White é intrepertado por Bryan Cranston que atuou nas seguintes séries: Breaking Bad.
    </h2>
  </div>
</styled.div>
`;
